### DISCUSSION TITLE
How to pass the global log id?

### DISCUSSION THREAD

**Discussion by 1798972** [Kong/kong#9112] [Help] [answered]
There are many plug-ins applied to a route, and it is difficult to find out which logs are related without log id.
How can you achieve the following effect like java? Mark a complete request processing?
![1658109734787_FDA9E004-9506-4ae7-B510-07B164E2620A](https://user-images.githubusercontent.com/49218070/179435925-471fb138-919a-443a-be90-ce736154e9a4.png)

**Comment by 1798972** [comment]
Can someone help me?

**Comment by chronolaw** [comment, marked as answer]
Maybe you could use this plugin: https://docs.konghq.com/hub/kong-inc/correlation-id/

**Reply by 1798972** [reply]
Thank you
Well, I thought about sending some information through http request header before.
However, in this case, it is troublesome to manually add the printed id to each log.
Is there any way to automatically process it in the program, similar to the java log framework above, and automatically add it to each log in a request?

**Reply by chronolaw** [reply]
I think you should customize Nginx's log format, there is a description in correlation-id's page.

**Reply by 1798972** [reply]
Sounds feasible. Let me verify it. Thank you very much! !

**Reply by 1798972** [reply]
Sorry for not reply,Thanks again.
Using variables, this operation is feasible in `access.log` Is there any way to show it in `error.log`, the running log of the plugin?

**Reply by chronolaw** [reply]
Now We can not customize Nginx's error log format, the only way to output info into error.log is Lua API `ngx.log` or `kong.log`.

**Reply by 1798972** [reply]
OK,I see.
thank you again for your help.
